# libdom: html
# 59 files, 259 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN upstream/libdom/src/html/html_anchor_element.h browser/libdom/src/html/html_anchor_element.h
--- upstream/libdom/src/html/html_anchor_element.h
+++ browser/libdom/src/html/html_anchor_element.h
@@ -11,7 +11,7 @@
 
 #include <dom/html/html_anchor_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_anchor_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_applet_element.h browser/libdom/src/html/html_applet_element.h
--- upstream/libdom/src/html/html_applet_element.h
+++ browser/libdom/src/html/html_applet_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_applet_element_h_
 
 #include <dom/html/html_applet_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_applet_element {
diff -urN upstream/libdom/src/html/html_area_element.h browser/libdom/src/html/html_area_element.h
--- upstream/libdom/src/html/html_area_element.h
+++ browser/libdom/src/html/html_area_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_area_element_h_
 
 #include <dom/html/html_area_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_area_element {
diff -urN upstream/libdom/src/html/html_base_element.h browser/libdom/src/html/html_base_element.h
--- upstream/libdom/src/html/html_base_element.h
+++ browser/libdom/src/html/html_base_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_base_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_base_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_basefont_element.h browser/libdom/src/html/html_basefont_element.h
--- upstream/libdom/src/html/html_basefont_element.h
+++ browser/libdom/src/html/html_basefont_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_basefont_element_h_
 
 #include <dom/html/html_basefont_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_base_font_element {
diff -urN upstream/libdom/src/html/html_body_element.h browser/libdom/src/html/html_body_element.h
--- upstream/libdom/src/html/html_body_element.h
+++ browser/libdom/src/html/html_body_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_body_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_body_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_br_element.h browser/libdom/src/html/html_br_element.h
--- upstream/libdom/src/html/html_br_element.h
+++ browser/libdom/src/html/html_br_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_br_element_h_
 
 #include <dom/html/html_br_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_br_element {
diff -urN upstream/libdom/src/html/html_button_element.h browser/libdom/src/html/html_button_element.h
--- upstream/libdom/src/html/html_button_element.h
+++ browser/libdom/src/html/html_button_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_button_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_button_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_canvas_element.h browser/libdom/src/html/html_canvas_element.h
--- upstream/libdom/src/html/html_canvas_element.h
+++ browser/libdom/src/html/html_canvas_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_canvas_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_canvas_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_directory_element.h browser/libdom/src/html/html_directory_element.h
--- upstream/libdom/src/html/html_directory_element.h
+++ browser/libdom/src/html/html_directory_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_directory_element_h_
 
 #include <dom/html/html_directory_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_directory_element {
diff -urN upstream/libdom/src/html/html_div_element.h browser/libdom/src/html/html_div_element.h
--- upstream/libdom/src/html/html_div_element.h
+++ browser/libdom/src/html/html_div_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_div_element_h_
 
 #include <dom/html/html_div_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_div_element {
diff -urN upstream/libdom/src/html/html_dlist_element.h browser/libdom/src/html/html_dlist_element.h
--- upstream/libdom/src/html/html_dlist_element.h
+++ browser/libdom/src/html/html_dlist_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_dlist_element_h_
 
 #include <dom/html/html_dlist_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_dlist_element {
diff -urN upstream/libdom/src/html/html_document.h browser/libdom/src/html/html_document.h
--- upstream/libdom/src/html/html_document.h
+++ browser/libdom/src/html/html_document.h
@@ -11,7 +11,7 @@
 #include <dom/html/html_document.h>
 #include <dom/html/html_element.h>
 
-#include "core/document.h"
+#include "core/document.h"
 
 /**
  * The dom_html_document class
diff -urN upstream/libdom/src/html/html_element.h browser/libdom/src/html/html_element.h
--- upstream/libdom/src/html/html_element.h
+++ browser/libdom/src/html/html_element.h
@@ -11,7 +11,7 @@
 #include <dom/html/html_element.h>
 #include <dom/html/html_elements.h>
 
-#include "core/element.h"
+#include "core/element.h"
 
 struct dom_html_document;
 
diff -urN upstream/libdom/src/html/html_fieldset_element.h browser/libdom/src/html/html_fieldset_element.h
--- upstream/libdom/src/html/html_fieldset_element.h
+++ browser/libdom/src/html/html_fieldset_element.h
@@ -11,7 +11,7 @@
 
 #include <dom/html/html_fieldset_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_field_set_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_font_element.h browser/libdom/src/html/html_font_element.h
--- upstream/libdom/src/html/html_font_element.h
+++ browser/libdom/src/html/html_font_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_font_element_h_
 
 #include <dom/html/html_font_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_font_element {
diff -urN upstream/libdom/src/html/html_form_element.h browser/libdom/src/html/html_form_element.h
--- upstream/libdom/src/html/html_form_element.h
+++ browser/libdom/src/html/html_form_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_form_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_collection;
 
diff -urN upstream/libdom/src/html/html_frame_element.h browser/libdom/src/html/html_frame_element.h
--- upstream/libdom/src/html/html_frame_element.h
+++ browser/libdom/src/html/html_frame_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_frame_element_h_
 
 #include <dom/html/html_frame_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_frame_element {
diff -urN upstream/libdom/src/html/html_frameset_element.h browser/libdom/src/html/html_frameset_element.h
--- upstream/libdom/src/html/html_frameset_element.h
+++ browser/libdom/src/html/html_frameset_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_frame_set_element_h_
 
 #include <dom/html/html_frameset_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_frame_set_element {
diff -urN upstream/libdom/src/html/html_head_element.h browser/libdom/src/html/html_head_element.h
--- upstream/libdom/src/html/html_head_element.h
+++ browser/libdom/src/html/html_head_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_head_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_head_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_heading_element.h browser/libdom/src/html/html_heading_element.h
--- upstream/libdom/src/html/html_heading_element.h
+++ browser/libdom/src/html/html_heading_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_heading_element_h_
 
 #include <dom/html/html_heading_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_heading_element {
diff -urN upstream/libdom/src/html/html_hr_element.h browser/libdom/src/html/html_hr_element.h
--- upstream/libdom/src/html/html_hr_element.h
+++ browser/libdom/src/html/html_hr_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_hr_element_h_
 
 #include <dom/html/html_hr_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_hr_element {
diff -urN upstream/libdom/src/html/html_html_element.h browser/libdom/src/html/html_html_element.h
--- upstream/libdom/src/html/html_html_element.h
+++ browser/libdom/src/html/html_html_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_html_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_html_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_iframe_element.h browser/libdom/src/html/html_iframe_element.h
--- upstream/libdom/src/html/html_iframe_element.h
+++ browser/libdom/src/html/html_iframe_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_iframe_element_h_
 
 #include <dom/html/html_iframe_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_iframe_element {
diff -urN upstream/libdom/src/html/html_image_element.h browser/libdom/src/html/html_image_element.h
--- upstream/libdom/src/html/html_image_element.h
+++ browser/libdom/src/html/html_image_element.h
@@ -11,7 +11,7 @@
 
 #include <dom/html/html_image_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_image_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_input_element.h browser/libdom/src/html/html_input_element.h
--- upstream/libdom/src/html/html_input_element.h
+++ browser/libdom/src/html/html_input_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_input_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_input_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_isindex_element.h browser/libdom/src/html/html_isindex_element.h
--- upstream/libdom/src/html/html_isindex_element.h
+++ browser/libdom/src/html/html_isindex_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_isindex_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_isindex_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_label_element.h browser/libdom/src/html/html_label_element.h
--- upstream/libdom/src/html/html_label_element.h
+++ browser/libdom/src/html/html_label_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_label_element_h_
 
 #include <dom/html/html_label_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_label_element {
diff -urN upstream/libdom/src/html/html_legend_element.h browser/libdom/src/html/html_legend_element.h
--- upstream/libdom/src/html/html_legend_element.h
+++ browser/libdom/src/html/html_legend_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_legend_element_h_
 
 #include <dom/html/html_legend_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_legend_element {
diff -urN upstream/libdom/src/html/html_li_element.h browser/libdom/src/html/html_li_element.h
--- upstream/libdom/src/html/html_li_element.h
+++ browser/libdom/src/html/html_li_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_li_element_h_
 
 #include <dom/html/html_li_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_li_element {
diff -urN upstream/libdom/src/html/html_link_element.h browser/libdom/src/html/html_link_element.h
--- upstream/libdom/src/html/html_link_element.h
+++ browser/libdom/src/html/html_link_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_link_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_link_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_map_element.h browser/libdom/src/html/html_map_element.h
--- upstream/libdom/src/html/html_map_element.h
+++ browser/libdom/src/html/html_map_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_map_element_h_
 
 #include <dom/html/html_map_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_map_element {
diff -urN upstream/libdom/src/html/html_menu_element.h browser/libdom/src/html/html_menu_element.h
--- upstream/libdom/src/html/html_menu_element.h
+++ browser/libdom/src/html/html_menu_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_menu_element_h_
 
 #include <dom/html/html_menu_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_menu_element {
diff -urN upstream/libdom/src/html/html_meta_element.h browser/libdom/src/html/html_meta_element.h
--- upstream/libdom/src/html/html_meta_element.h
+++ browser/libdom/src/html/html_meta_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_meta_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_meta_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_mod_element.h browser/libdom/src/html/html_mod_element.h
--- upstream/libdom/src/html/html_mod_element.h
+++ browser/libdom/src/html/html_mod_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_mod_element_h_
 
 #include <dom/html/html_mod_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_mod_element {
diff -urN upstream/libdom/src/html/html_object_element.h browser/libdom/src/html/html_object_element.h
--- upstream/libdom/src/html/html_object_element.h
+++ browser/libdom/src/html/html_object_element.h
@@ -11,7 +11,7 @@
 
 #include <dom/html/html_object_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_object_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_olist_element.h browser/libdom/src/html/html_olist_element.h
--- upstream/libdom/src/html/html_olist_element.h
+++ browser/libdom/src/html/html_olist_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_olist_element_h_
 
 #include <dom/html/html_olist_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_olist_element {
diff -urN upstream/libdom/src/html/html_opt_group_element.h browser/libdom/src/html/html_opt_group_element.h
--- upstream/libdom/src/html/html_opt_group_element.h
+++ browser/libdom/src/html/html_opt_group_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_opt_group_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_opt_group_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_option_element.h browser/libdom/src/html/html_option_element.h
--- upstream/libdom/src/html/html_option_element.h
+++ browser/libdom/src/html/html_option_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_option_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_option_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_options_collection.h browser/libdom/src/html/html_options_collection.h
--- upstream/libdom/src/html/html_options_collection.h
+++ browser/libdom/src/html/html_options_collection.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_options_collection.h>
 
-#include "html/html_collection.h"
+#include "html/html_collection.h"
 
 struct dom_node_internal;
 
diff -urN upstream/libdom/src/html/html_paragraph_element.h browser/libdom/src/html/html_paragraph_element.h
--- upstream/libdom/src/html/html_paragraph_element.h
+++ browser/libdom/src/html/html_paragraph_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_paragraph_element_h_
 
 #include <dom/html/html_paragraph_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_paragraph_element {
diff -urN upstream/libdom/src/html/html_param_element.h browser/libdom/src/html/html_param_element.h
--- upstream/libdom/src/html/html_param_element.h
+++ browser/libdom/src/html/html_param_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_param_element_h_
 
 #include <dom/html/html_param_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_param_element {
diff -urN upstream/libdom/src/html/html_pre_element.h browser/libdom/src/html/html_pre_element.h
--- upstream/libdom/src/html/html_pre_element.h
+++ browser/libdom/src/html/html_pre_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_pre_element_h_
 
 #include <dom/html/html_pre_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_pre_element {
diff -urN upstream/libdom/src/html/html_quote_element.h browser/libdom/src/html/html_quote_element.h
--- upstream/libdom/src/html/html_quote_element.h
+++ browser/libdom/src/html/html_quote_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_quote_element_h_
 
 #include <dom/html/html_quote_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_quote_element {
diff -urN upstream/libdom/src/html/html_script_element.h browser/libdom/src/html/html_script_element.h
--- upstream/libdom/src/html/html_script_element.h
+++ browser/libdom/src/html/html_script_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_script_element_h_
 
 #include <dom/html/html_script_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_script_element {
diff -urN upstream/libdom/src/html/html_select_element.h browser/libdom/src/html/html_select_element.h
--- upstream/libdom/src/html/html_select_element.h
+++ browser/libdom/src/html/html_select_element.h
@@ -10,8 +10,8 @@
 
 #include <dom/html/html_select_element.h>
 
-#include "html/html_element.h"
-#include "html/html_options_collection.h"
+#include "html/html_element.h"
+#include "html/html_options_collection.h"
 
 struct dom_html_select_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_style_element.h browser/libdom/src/html/html_style_element.h
--- upstream/libdom/src/html/html_style_element.h
+++ browser/libdom/src/html/html_style_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_style_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_style_element {
 	struct dom_html_element base;
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libdom/src/html/html_table_element.c browser/libdom/src/html/html_table_element.c
--- upstream/libdom/src/html/html_table_element.c	2026-09-20 23:27:12.698490669 +0100
+++ browser/libdom/src/html/html_table_element.c	2026-09-20 23:19:17.329617272 +0100
@@ -10,16 +10,16 @@
 
 #include <dom/html/html_table_element.h>
 
-#include "html/html_document.h"
-#include "html/html_table_element.h"
-#include "html/html_tablecaption_element.h"
-#include "html/html_tablesection_element.h"
-#include "html/html_tablerow_element.h"
-#include "html/html_collection.h"
-
-#include "core/node.h"
-#include "core/attr.h"
-#include "utils/utils.h"
+#include "html/html_document.h"
+#include "html/html_table_element.h"
+#include "html/html_tablecaption_element.h"
+#include "html/html_tablesection_element.h"
+#include "html/html_tablerow_element.h"
+#include "html/html_collection.h"
+
+#include "core/node.h"
+#include "core/attr.h"
+#include "utils/utils.h"
 
 static const struct dom_element_protected_vtable _protect_vtable = {
 	{
@@ -464,13 +464,12 @@
 		dom_html_document *doc = (dom_html_document *)
 				((dom_node_internal *) element)->owner;
 
-		struct dom_html_element_create_params params = {
-			.type = DOM_HTML_ELEMENT_TYPE_CAPTION,
-			.doc = doc,
-			.name = doc->elements[DOM_HTML_ELEMENT_TYPE_CAPTION],
-			.namespace = ((dom_node_internal *)element)->namespace,
-			.prefix = ((dom_node_internal *)element)->prefix
-		};
+		struct dom_html_element_create_params params;
+		params.type = DOM_HTML_ELEMENT_TYPE_CAPTION;
+		params.doc = doc;
+		params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_CAPTION];
+		params.namespace = ((dom_node_internal *)element)->namespace;
+		params.prefix = ((dom_node_internal *)element)->prefix;
 
 		exp = _dom_html_table_caption_element_create(&params,
 				(dom_html_table_caption_element **)caption);
@@ -537,13 +536,12 @@
 		dom_html_document *doc = (dom_html_document *)
 				((dom_node_internal *) element)->owner;
 
-		struct dom_html_element_create_params params = {
-			.type = DOM_HTML_ELEMENT_TYPE_TFOOT,
-			.doc = doc,
-			.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TFOOT],
-			.namespace = ((dom_node_internal *)element)->namespace,
-			.prefix = ((dom_node_internal *)element)->prefix
-		};
+		struct dom_html_element_create_params params;
+		params.type = DOM_HTML_ELEMENT_TYPE_TFOOT;
+		params.doc = doc;
+		params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TFOOT];
+		params.namespace = ((dom_node_internal *)element)->namespace;
+		params.prefix = ((dom_node_internal *)element)->prefix;
 
 		exp = _dom_html_table_section_element_create(&params,
 				(dom_html_table_section_element **)t_foot);
@@ -615,13 +613,12 @@
 		dom_html_document *doc = (dom_html_document *)
 				((dom_node_internal *) element)->owner;
 
-		struct dom_html_element_create_params params = {
-			.type = DOM_HTML_ELEMENT_TYPE_THEAD,
-			.doc = doc,
-			.name = doc->elements[DOM_HTML_ELEMENT_TYPE_THEAD],
-			.namespace = ((dom_node_internal *)element)->namespace,
-			.prefix = ((dom_node_internal *)element)->prefix
-		};
+		struct dom_html_element_create_params params;
+		params.type = DOM_HTML_ELEMENT_TYPE_THEAD;
+		params.doc = doc;
+		params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_THEAD];
+		params.namespace = ((dom_node_internal *)element)->namespace;
+		params.prefix = ((dom_node_internal *)element)->prefix;
 
 		exp = _dom_html_table_section_element_create(&params,
 				(dom_html_table_section_element **)t_head);
@@ -698,13 +695,12 @@
 		dom_html_document *doc = (dom_html_document *)
 				((dom_node_internal *) element)->owner;
 
-		struct dom_html_element_create_params params = {
-			.type = DOM_HTML_ELEMENT_TYPE_TBODY,
-			.doc = doc,
-			.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TBODY],
-			.namespace = ((dom_node_internal *)element)->namespace,
-			.prefix = ((dom_node_internal *)element)->prefix
-		};
+		struct dom_html_element_create_params params;
+		params.type = DOM_HTML_ELEMENT_TYPE_TBODY;
+		params.doc = doc;
+		params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TBODY];
+		params.namespace = ((dom_node_internal *)element)->namespace;
+		params.prefix = ((dom_node_internal *)element)->prefix;
 
 		exp = _dom_html_table_section_element_create(&params, t_body);
 		if(exp != DOM_NO_ERR) {
@@ -762,13 +758,12 @@
 		dom_html_table_row_element *row;
 		dom_node *new_row;
 
-		struct dom_html_element_create_params params = {
-			.type = DOM_HTML_ELEMENT_TYPE_TR,
-			.doc = doc,
-			.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TR],
-			.namespace = ((dom_node_internal *)element)->namespace,
-			.prefix = ((dom_node_internal *)element)->prefix
-		};
+		struct dom_html_element_create_params params;
+		params.type = DOM_HTML_ELEMENT_TYPE_TR;
+		params.doc = doc;
+		params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TR];
+		params.namespace = ((dom_node_internal *)element)->namespace;
+		params.prefix = ((dom_node_internal *)element)->prefix;
 
 		exp = _dom_html_table_row_element_create(&params, &row);
 		if(exp != DOM_NO_ERR) {
diff -urN upstream/libdom/src/html/html_table_element.h browser/libdom/src/html/html_table_element.h
--- upstream/libdom/src/html/html_table_element.h
+++ browser/libdom/src/html/html_table_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_table_element_h_
 
 #include <dom/html/html_table_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_table_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_tablecaption_element.h browser/libdom/src/html/html_tablecaption_element.h
--- upstream/libdom/src/html/html_tablecaption_element.h
+++ browser/libdom/src/html/html_tablecaption_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_table_caption_element_h_
 
 #include <dom/html/html_tablecaption_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_table_caption_element {
diff -urN upstream/libdom/src/html/html_tablecell_element.h browser/libdom/src/html/html_tablecell_element.h
--- upstream/libdom/src/html/html_tablecell_element.h
+++ browser/libdom/src/html/html_tablecell_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_table_cell_element_h_
 
 #include <dom/html/html_tablecell_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_table_cell_element {
diff -urN upstream/libdom/src/html/html_tablecol_element.h browser/libdom/src/html/html_tablecol_element.h
--- upstream/libdom/src/html/html_tablecol_element.h
+++ browser/libdom/src/html/html_tablecol_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_table_col_element_h_
 
 #include <dom/html/html_tablecol_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_table_col_element {
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libdom/src/html/html_tablerow_element.c browser/libdom/src/html/html_tablerow_element.c
--- upstream/libdom/src/html/html_tablerow_element.c	2026-09-20 23:27:12.698490669 +0100
+++ browser/libdom/src/html/html_tablerow_element.c	2026-09-20 23:19:17.330711222 +0100
@@ -12,13 +12,13 @@
 #include <dom/html/html_tablerow_element.h>
 #include <dom/html/html_table_element.h>
 
-#include "html/html_document.h"
-#include "html/html_tablerow_element.h"
-#include "html/html_collection.h"
-
-#include "core/node.h"
-#include "core/attr.h"
-#include "utils/utils.h"
+#include "html/html_document.h"
+#include "html/html_tablerow_element.h"
+#include "html/html_collection.h"
+
+#include "core/node.h"
+#include "core/attr.h"
+#include "utils/utils.h"
 
 static const struct dom_element_protected_vtable _protect_vtable = {
 	{
@@ -374,13 +374,12 @@
 	uint32_t len; 			/*< The size of the cell collection */
 	dom_exception exp;		/*< Variable for getting the exceptions*/
 
-	struct dom_html_element_create_params params = {
-		.type = DOM_HTML_ELEMENT_TYPE_TD,
-		.doc = doc,
-		.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TD],
-		.namespace = ((dom_node_internal *)element)->namespace,
-		.prefix = ((dom_node_internal *)element)->prefix
-	};
+	struct dom_html_element_create_params params;
+	params.type = DOM_HTML_ELEMENT_TYPE_TD;
+	params.doc = doc;
+	params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TD];
+	params.namespace = ((dom_node_internal *)element)->namespace;
+	params.prefix = ((dom_node_internal *)element)->prefix;
 
 	exp = _dom_html_element_create(&params, &new_cell);
 	if (exp != DOM_NO_ERR)
diff -urN upstream/libdom/src/html/html_tablerow_element.h browser/libdom/src/html/html_tablerow_element.h
--- upstream/libdom/src/html/html_tablerow_element.h
+++ browser/libdom/src/html/html_tablerow_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_table_row_element_h_
 
 #include <dom/html/html_tablerow_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_table_row_element {
 	struct dom_html_element base;
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libdom/src/html/html_tablesection_element.c browser/libdom/src/html/html_tablesection_element.c
--- upstream/libdom/src/html/html_tablesection_element.c	2026-09-20 23:27:12.698490669 +0100
+++ browser/libdom/src/html/html_tablesection_element.c	2026-09-20 23:19:17.331372489 +0100
@@ -10,16 +10,16 @@
 
 #include <dom/html/html_tablesection_element.h>
 
-#include "html/html_document.h"
-#include "html/html_tablesection_element.h"
-#include "html/html_tablerow_element.h"
-
-#include "html/html_collection.h"
-#include "html/html_element.h"
-
-#include "core/node.h"
-#include "core/attr.h"
-#include "utils/utils.h"
+#include "html/html_document.h"
+#include "html/html_tablesection_element.h"
+#include "html/html_tablerow_element.h"
+
+#include "html/html_collection.h"
+#include "html/html_element.h"
+
+#include "core/node.h"
+#include "core/attr.h"
+#include "utils/utils.h"
 
 static const struct dom_element_protected_vtable _protect_vtable = {
 	{
@@ -237,13 +237,12 @@
 	dom_exception exp;		/*< Variable for getting the exceptions*/
 	dom_node *new_node;
 
-	struct dom_html_element_create_params params = {
-		.type = DOM_HTML_ELEMENT_TYPE_TR,
-		.doc = doc,
-		.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TR],
-		.namespace = ((dom_node_internal *)element)->namespace,
-		.prefix = ((dom_node_internal *)element)->prefix
-	};
+	struct dom_html_element_create_params params;
+	params.type = DOM_HTML_ELEMENT_TYPE_TR;
+	params.doc = doc;
+	params.name = doc->elements[DOM_HTML_ELEMENT_TYPE_TR];
+	params.namespace = ((dom_node_internal *)element)->namespace;
+	params.prefix = ((dom_node_internal *)element)->prefix;
 
 	exp = _dom_html_table_row_element_create(&params,
 			(dom_html_table_row_element **)(void *) &new_node);
diff -urN upstream/libdom/src/html/html_tablesection_element.h browser/libdom/src/html/html_tablesection_element.h
--- upstream/libdom/src/html/html_tablesection_element.h
+++ browser/libdom/src/html/html_tablesection_element.h
@@ -11,7 +11,7 @@
 
 #include <dom/html/html_tablesection_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_table_section_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_text_area_element.h browser/libdom/src/html/html_text_area_element.h
--- upstream/libdom/src/html/html_text_area_element.h
+++ browser/libdom/src/html/html_text_area_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_text_area_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_text_area_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_title_element.h browser/libdom/src/html/html_title_element.h
--- upstream/libdom/src/html/html_title_element.h
+++ browser/libdom/src/html/html_title_element.h
@@ -10,7 +10,7 @@
 
 #include <dom/html/html_title_element.h>
 
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 struct dom_html_title_element {
 	struct dom_html_element base;
diff -urN upstream/libdom/src/html/html_ulist_element.h browser/libdom/src/html/html_ulist_element.h
--- upstream/libdom/src/html/html_ulist_element.h
+++ browser/libdom/src/html/html_ulist_element.h
@@ -10,7 +10,7 @@
 #define dom_internal_html_u_list_element_h_
 
 #include <dom/html/html_ulist_element.h>
-#include "html/html_element.h"
+#include "html/html_element.h"
 
 
 struct dom_html_u_list_element {
